packages/nitro-protocol/contracts/test/TestEthAssetHolder.sol

```diff
@@ -2,12 +2,22 @@
 pragma solidity 0.7.4;
 pragma experimental ABIEncoderV2;
 import '../ETHAssetHolder.sol';
+import './TESTAssetHolder.sol';
 
 /**
  * @dev This contract is a clone of the ETHAssetHolder contract. It will be initialized to point to the TestNitroAdjudicator.
  */
 contract TestEthAssetHolder is ETHAssetHolder {
     uint256 public dummy; // this is simply to make the contract have distinct bytecode from the ETHAssetHolder (otherwise bytecode verification can fail)
 
+    /**
+     * @dev Manually set the holdings mapping to a given amount for a given channelId.  Shortcuts the deposit workflow (ONLY USE IN A TESTING ENVIRONMENT)
+     * @param channelId Unique identifier for a state channel.
+     * @param amount The number of assets that should now be "escrowed: against channelId
+     */
+    function setHoldings(bytes32 channelId, uint256 amount) external {
+        holdings[channelId] = amount;
+    }
```

Comment:
I don't really understand why this change makes it a "full eth asset holder".

Reply:
This isn't. This is just a change that was necessary to use the full eth asset holder in the current test. I have another PR in the pipe that will actually do away with this contract altogether.